src/sonic-yang-models/yang-models/sonic-vlan-sub-interface.yang

```diff
@@ -36,9 +36,11 @@ module sonic-vlan-sub-interface {
                 key "name";
 
                 leaf name {
-                    must "substring-before(current(), '.') = /port:sonic-port/port:PORT/port:PORT_LIST/port:name"
+                    must "(substring-before(current(), '.') = /port:sonic-port/port:PORT/port:PORT_LIST/port:name) or " +
+                         "(starts-with(substring-before(current(), '.'), 'Eth') and " +
+                         "concat('Ethernet', substring-after(substring-before(current(), '.'), 'Eth')) = /port:sonic-port/port:PORT/port:PORT_LIST/port:name)"
```

Comment:
shouldn't the left side be concat('Ethernet', substring-after(current(), '.'))

Reply:
` substring-after(current(), '.')` will return the vlan id, what we need is the substring between `Eth` and `.`, which contains the port index, right?

Reply:
My bad. you are right